src/history/hs.c

```diff
@@ -512,6 +491,52 @@ __hs_insert_record_with_btree(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BT
     while ((ret = __hs_insert_record_with_btree_int(
               session, cursor, btree, key, upd, type, hs_value, stop_time_point)) == WT_RESTART)
         ;
```

Comment:
You didn't add this code, but it would be interesting to increment a statistic here. Busy spinning loops make me nervous.

Reply:
3c69ecc